rescript-relay-documentation/docs/tutorial/3-query-basics.md

```diff
@@ -0,0 +1,324 @@
+
+```rescript
+// color1
+module NewsfeedQuery = %relay(`
+  // color2
+  query NewsfeedQuery {
+    topStory {
+      // color3
+      title
+      // color3
+      summary
+      // color4
+      poster {
+        name
+        profilePicture {
+          url
+        }
+      }
+      thumbnail {
+        url
+      }
+    }
+  }
+`)
+```
+
+Let’s break this down:
+
+- To embed GraphQL within ReScript, we write a string literal <span class="color1">marked with the <code>%relay(``)</code> tag</span>. This tag allows the Relay compiler to find and compile the GraphQL within a ReScript codebase.
+- Our GraphQL string consists of a <span class="color2">query declaration</span> with the keyword `query` and then a query name.
+- Inside the query declaration are _fields_, which specify what information to query for*:*
+  - Some fields are _<span class="color3">scalar fields</span>_ that retrieve a string, number, or other unit of information.
+  - Other fields are _<span class="color4">edges</span>_ that let us traverse from one node in the graph to another. When a field is an edge, it’s followed by another block `{ }` containing fields for the node at the other end of the edge. Here, the `poster` field is an edge that goes from a Story to a Person who posted it. Once we’ve traversed to the Person, we can include fields about the Person such as their `name`.
+- Finally, while you can call the `module` anything you like, the query name (i.e. the stuff that comes right after `query`) must follow relay conventions. A query must start with the name of the component in which it is used and end in "query". That means valid names for this query is e.g. `NewsfeedQuery` and `NewsfeedTopStoryQuery`, but not `MainNewsfeedQuery` or `NewsfeedTopStory`. Don't worry about remembering these rules, the compiler will tell you!
+
+This illustrates the part of the graph that this query is asking for:
+
+![Parts of the GraphQL query](/img/docs/tutorial/query-breakdown.png)
+
+Now that we’ve defined the query, we need to modify our React component to fetch it and to use the data returned by the server.
+
+Turn back to the `Newsfeed` component and start by deleting the placeholder data. Then replace it with this (which is supposed to not compile, so don't panic when you see the errors!):
+
+```rescript
+@react.component
+let make = () => {
+  let data = NewsfeedQuery.use(~variables=(), ())
+
+  switch data.topStory {
+  | None => React.null
+  | Some(topStory) =>
+    <div className="newsfeed">
+      <Story story={(topStory :> Story.story)} />
+    </div>
+  }
+}
+```
+
+`topStory` returned by the query has type `NewsfeedQuery_graphql.Types.response_topStory`. Since this is compatible with the type `Story.story` expected by the `story` prop, we can use `:>` is needed to convert `topStory` into the expected type. This is necessary because ReScript uses structural types. See the Deep Dive below for more.
```

Comment:
```suggestion
`topStory` returned by the query has type `NewsfeedQuery_graphql.Types.response_topStory`. Since this is compatible with the type `Story.story` expected by the `story` prop, we can use `:>` is needed to convert `topStory` into the expected type. This is necessary because ReScript uses nominal types. See the Deep Dive below for more.
```